Mathlib/AlgebraicGeometry/EllipticCurve/Weierstrass.lean

```diff
@@ -35,6 +35,7 @@ splitting field of `R` are precisely the $X$-coordinates of the non-zero 2-torsi
 
  * `WeierstrassCurve`: a Weierstrass curve over a commutative ring.
  * `WeierstrassCurve.Δ`: the discriminant of a Weierstrass curve.
+ * `WeierstrassCurve.VariableChange`: a change of variable of Weierstrass curves.
```

Comment:
"change of variable*s*"? I've been using the plural form everywhere below. I'm also unsure if we should call it `VariableChange` since it might be confused with `variableChange`, but I don't have a better name either.

Reply:
Yes, it's called change of variable**s** in Silverman's book. I think the function to change the curve could be called `changeCurve` (at least in my previous attempt).

Reply:
What about `changeCoordinate`? I'm trying to compare it with `baseChange` (which should probably be called `changeBase` if we decide to rename everything), where it's clear what we're changing (whereas `changeCurve` doesn't say exactly what).

Reply:
My proposal `changeCurve` is in fact coming from the names of corresponding functions in Pari/GP: `ellchangecurve()`, `ellchangepoint()` and `ellchangepointinv()`.
```
ellchangecurve(E,v):

   Changes  the data for the elliptic curve E by changing the coordinates using
the vector v = [u,r,s,t],  i.e. if x' and y' are the new coordinates,  then x =
u^2x'+r, y = u^3y'+su^2x'+t.   E must be an ell structure as output by ellinit.
The  special case v = 1 is also used instead of [1,0,0,0] to denote the trivial
coordinate change.

ellchangepoint(x,v):

   Changes  the coordinates of the point or vector of points x using the vector
v = [u,r,s,t], i.e. if x' and y' are the new coordinates, then x = u^2x'+r, y =
u^3y'+su^2x'+t (see also ellchangecurve).

   ? E0 = ellinit([1,1]); P0 = [0,1]; v = [1,2,3,4];
   ? E = ellchangecurve(E0, v);
   ? P = ellchangepoint(P0,v)
   %3 = [-2, 3]
   ? ellisoncurve(E, P)
   %4 = 1
   ? ellchangepointinv(P,v)
   %5 = [0, 1]
```
BTW, I think `changeRing` is better than `changeBase` (which sounds too weird), while I think `baseChange` is the best, since it is a standard word in mathematics.

Reply:
I see - let's do `changeCurve` and `changeRing` then